buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/performance/PerformanceTestPlugin.kt

```diff
@@ -103,6 +106,43 @@ class PerformanceTestPlugin : Plugin<Project> {
         tasks.register("performanceReport")
     }
 
+    private
+    fun Project.registerForkPointDistributionTask() {
+        whenNotOnMasterOrReleaseBranch {
+            val buildForkPointDistribution = tasks.register("buildForkPointDistribution", BuildForkPointDistribution::class) {
+                dependsOn("determineForkPoint")
+            }
+            tasks.register("determineForkPoint") {
+                doLast {
+                    val masterForkPointCommit = execAndGetStdout("git", "merge-base", "master", "HEAD")
+                    val releaseForkPointCommit = execAndGetStdout("git", "merge-base", "release", "HEAD")
+                    val forkPointCommit =
+                        if (exec { isIgnoreExitValue = true; commandLine("git", "merge-base", "--is-ancestor", masterForkPointCommit, releaseForkPointCommit) }.exitValue == 0)
+                            releaseForkPointCommit
+                        else
+                            masterForkPointCommit
+                    buildForkPointDistribution.get().forkPointCommitId = forkPointCommit.substring(0, 7)
```

Comment:
My intention was to make the `forkPointCommit` an input of  `buildForkPointDistribution` so it can be cached, so anyway we need an `@Input forkPointCommit` property exposed in `buildForkPointDistribution`, why don't make it in `buildForkPointDistribution`?

Reply:
After thinking a little bit more, I think that both tasks should have a `Property<String>` for the `forkPointCommit`. The property is an input to `buildForkPointDistribution` and set in the task action of `determineForkPoint`. So it is somewhat an output property of `determineForkPoint` - but we don't have that modelled yet, so it would be annotated with `@Internal`.

The plugin would then wire the two properties together at _configuration_ time, which works since both are lazy - i.e. `Property`s. Then there is not need to update properties of other tasks  during task actions. I think that pattern should be avoided - tasks should not reach into other tasks. The plugin should take care of the wiring.